src/middlewares/auth.middleware.js

```diff
@@ -24,7 +23,61 @@ const ensureUniqueUsername = async (req, res, next) => {
     }
 };
 
+
+const findUser = async(req, res, next) =>{
+    try
+    {
+        const user = await User.findOne({email: req.body.email}).exec();
```

Comment:
Here you need to select password.

Reply:
But then if it is a wrong password, I won't be able to detect it correctly as I will get null user for both wrong password and user not found. Shouldn't they be different?

Reply:
you are right.